### PR TITLE
#164829363 Implement feature: Users create profile

### PR DESCRIPTION
**Pull request by Kyppy**
#### Description
The PR introduces the 'Profile' app to the project, which includes URL endpoints that allow the creation of, and edits to, the User profile.
Implements the following views together with their methods for the 'Users should be able to create their profiles' feature:

|  Method  | Endpoint |  Description |
|---|---|---|
| GET  | /api/profiles/:username/ | Get an existing user profile |
| POST  | /api/profile  | Create a new user profile  |
| PUT  | /api/profile/:username/edit/  | Edits an existing profile  |
| PATCH  | /api/profile/:username/avatar/  | Edits an existing avatar  |
#### Description of Task to be completed?
An authenticated user should be able to create, read, update their profile.
An authenticated user should be able to update their profile avatar.

*NOTE*
Authenticating as a user requires you to use the sign-up and login views.

1) Sign-up
Access the endpoint using: http://127.0.0.1:8000/api/users/
Example sign-up payload:
```
{
  "user": {
    "email": "demo@mail.com",
    "username" :"Mary",
    "password":abc123ok
  }
}
```

2) Login
Access the endpoint using: http://127.0.0.1:8000/api/users/login/
Example login payload:
```
{
  "user": {
    "email": "demo@mail.com",
    "password":abc123ok
  }
}
```
After a successful login, a token will be returned. Copy and include it in the authorization headers when attempting to access any of the 'profile' views.

#### How should this be manually tested?
* Clone repo to local machine
* 'Cd' into cloned repository folder
* Create a virtual environment by running `virtualenv venv`
* Activate the virtualenv `source venv\bin\activate`
* Install dependencies `pip install -r dev-requirements.txt`
* Run the Django server to serve up the app `python manage.py runserver`
###### Test with Postman using the payloads
* For **PUT** and **POST** endpoints
```
{
  "profile": {
    "user_bio": "I work at statefarm",
    "name" :"Mary Doe",
    "number_of_followers":78,
    "number_of_followings" : 50,
    "total_articles": 89,
  }
}
```
###### Upload an avatar
* Url Endpoint `PATCH /api/profile/<str:username>/avatar/`
On Postman
* Select the Body tab
* Ensure the parameter option is set to form-data
* Under the KEY field add the word `avatar` 
* When you hover on the far right side of the KEY field a dropdown will appear. Select File among the options in the dropdown
* In the VALUE field, press the Select Files button and choose an image file on the file picker that pops up.

#### How has this been tested?
- [x] Unittest
Run 'python manage.py test' to run the test and receive a report with test coverage.

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164829363

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,4 +1,31 @@
+astroid==2.2.5
```
Comment (WNjihia) [inline dev-requirements.txt]: Most of these dependencies should be in the requirements.txt. The `--r requirements.txt` file removed here allowed devs to run pip install -r dev-requirements.txt and have the dependencies in the requirements.txt installed.

Comment (Kyppy) [reply dev-requirements.txt]: Noted. I will correct that now.

**Review by Raywire — DISMISSED** [review]
Everything looks to be good please remember to add the .DS_Store file to your gitignore file.

**Review by ipaullly — DISMISSED** [review]
I like how you implemented custom HTTP status and error messages using the APIException base class in the EditProfile and Avatar views. the tests also seem comprehensive

**Review by WNjihia — DISMISSED** [review]

File: authors/settings.py [hunk 2]
```diff
@@ -159,8 +159,9 @@
     ),
 }
 
+
 cloudinary.config(
```
Comment (kirega) [inline authors/settings.py:163]: Do not forget to update travis and heroku with these env variables

**Review by kirega — APPROVED** [review]

**Review by WNjihia — APPROVED** [review]

**Pull request merged**